src/arch/ArchHooks/ArchHooks_Unix.cpp

```diff
@@ -128,21 +128,20 @@ namespace
 
 	void OpenGetTime()
 	{
+		// This is not thread-safe. Use std::call_once if it needs to be.
 		static bool bInitialized = false;
-		if( bInitialized )
-			return;
-		bInitialized = true;
-
-		/* Check whether the clock is actually supported. */
-		timespec ts;
-		if( clock_getres(CLOCK_MONOTONIC, &ts) == -1 )
-			return;
-
-		/* If the resolution is worse than a millisecond, fall back on CLOCK_REALTIME. */
-		if( ts.tv_sec > 0 || ts.tv_nsec > 1000000 )
-			return;
-
-		g_Clock = CLOCK_MONOTONIC;
+		if (!bInitialized) {
+			timespec ts;
+			// If CLOCK_MONOTONIC is available, set g_Clock to use it.
+			if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0) {
+				g_Clock = CLOCK_MONOTONIC;
+			}
+			else {
+				// Return, because g_Clock has not been changed from CLOCK_REALTIME.
```

Comment:
I think we need to drop this else block entirely so that it falls through to set `bInitialized` to true.

Actually looking at the old code, I don't actually see any issue with it, especially compared to what we're doing now.

Why do we need to change this?

Reply:
It seems like we found conflicting information about `CLOCK_MONOTONIC_RAW`. I had seen that it allowed for potentially higher resolution than `CLOCK_MONOTONIC` and this PR mostly existed to take advantage of the `RAW` clock if it's available.

If there isn't consensus that `CLOCK_MONOTONIC_RAW` is potentially higher precision, I wouldn't want to implement it.

I think we should use CLOCK_MONOTONIC if it's available, though, so maybe we can remove the check for resolution in case it provides an incorrect result and just use that clock if it's available.

Reply:
Why do we think `clock_getres` would provide an incorrect result?

Reply:
I think that sounds fine to me!

Reply:
I think replacing the old code with:

```cpp
void OpenGetTime()
{
		static bool bInitialized = false;
		if( bInitialized )
			return;
		bInitialized = true;

		/* Check whether the clock is actually supported. */
		timespec ts;
		if( clock_gettime(CLOCK_MONOTONIC, &ts) == -1 )
			return;

		g_Clock = CLOCK_MONOTONIC;
}
```

seems fine to me